### PR TITLE
chore(dependencies): update to spring boot 2 and kork BOM dependency management

### PR DESCRIPTION
**Pull request by cfieber**
This updates igor to use the kork BOM as the source of dependency opinions, and updates to spring boot 2.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -70,15 +55,6 @@ allprojects {
...
-
-    //c&p this because NetflixOss reverts it to 1.7 and ends up getting applied last..
-    project.plugins.withType(JavaBasePlugin) {
-        JavaPluginConvention convention = project.convention.getPlugin(JavaPluginConvention)
-        convention.sourceCompatibility = JavaVersion.VERSION_1_8
-        convention.targetCompatibility = JavaVersion.VERSION_1_8
-    }
 }
 
 defaultTasks 'run'
```
Comment (robzienert) [inline build.gradle]: I just love how the bom stuff cleans up the main gradle file. 🥓

File: igor-web/src/main/groovy/com/netflix/spinnaker/igor/Main.groovy [hunk 2]
```diff
@@ -48,6 +48,8 @@ class Main extends SpringBootServletInitializer {
     ]
 
     static {
+        System.setProperty("spring.main.allow-bean-definition-overriding", "true");
```
Comment (robzienert) [inline igor-web/src/main/groovy/com/netflix/spinnaker/igor/Main.groovy]: I don't think this is necessary anymore - I pulled this out into kork.

File: build.gradle (hunk 1 above)
Comment (cfieber) [reply build.gradle]: indeed, is niiice

File: igor-web/src/main/groovy/com/netflix/spinnaker/igor/Main.groovy (hunk 2 above)
Comment (cfieber) [reply igor-web/src/main/groovy/com/netflix/spinnaker/igor/Main.groovy]: good call, saw that in kork

**Pull request merged**